src/NetMPA.Catalog/NetMPA.Catalog.Api/Controllers/CatalogController.cs

```diff
@@ -0,0 +1,216 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using NetMPA.Catalog.Api.Controllers.Requests;
+using NetMPA.Catalog.Api.Models;
+using NetMPA.Catalog.Bll.Interfaces.Services;
+//using NetMPA.Catalog.Bll.Models;
+
+namespace NetMPA.Catalog.Api.Controllers
+{
+    /// <summary>
+    /// Catalog API
+    /// </summary>
+    [ApiVersion("1.0")]
+    [Route("api/v{version:apiVersion}/[controller]")]
+ 
+    [ApiController]
+    public class CatalogController : ControllerBase
+    {
+        private readonly ILogger<CatalogController> logger;
+        private readonly ICategoryService categoryService;
+        private readonly IProductService productService;
+        private readonly IMapper mapper;
+
+        public CatalogController(ICategoryService categoryService, IProductService productService,IMapper mapper, ILogger<CatalogController> logger)
+        {
+            this.logger = logger;
+            this.categoryService = categoryService;
+            this.productService = productService;
+            this.mapper = mapper;
+        }
+
+        /// <summary>
+        /// List of Categories
+        /// </summary>
+        [HttpGet()]
+        [Route("category")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<IEnumerable<Category>>> GetAllCategories([FromQuery] PagedRequest pagedRequest)
+        {
+            var result = await this.categoryService.List();
+            if (result == null) return NotFound();
+            return Ok(result.Take(pagedRequest.Take).Skip(pagedRequest.Skip).Select(this.mapper.Map<Category>));
```

Comment:
We usually do paging at DB level, not in controller. But controller is responsible for parameters validation (e.g. negative page size or index) and should return appropriate status code

Reply:
yes, sure